### ISSUE TITLE
Feedback Requested - Broker on by default

### ISSUE THREAD

**Issue by shoatman** [AzureAD/microsoft-authentication-library-for-android#639]
Hi Folks,

Looking for community feedback on the following:

There is a desire to enable the Microsoft authentication broker by default within MSAL.  The new app registration experience within the Azure Portal will lead you through creating a redirect URI that's compatible with the broker.  

For those who do not know the "Broker" is what Microsoft uses to enable:

- Cross app SSO
- Device Registration
- Compliance with conditional access policies including:
  - Device Policies (Mobile device management)
  - App Policies (Intune App Protection)

The broker library is an implementation of the Android Abstract Account Authenticator and includes a custom account called a "Work account".  The broker library ships as part of either the Microsoft Authenticator or the Intune Company Portal.  

If your app does not have a broker compatible redirect URI registered then the broker will not be used; however if you do have a broker enabled redirect URI registered then the broker would become the default.  

My question is: Would you like to have control over whether the broker is used or not?

Thanks

shane

**Comment by NLLAPPS** [comment]
Excuse my ignorance but, does that mean user will be able to authorize apps from Microsoft Authenticator?

DropBox Java SDK on Android works similarly, if user has Dropbox app installed and logged in, they authorize requests from Dropbox app.

**Comment by shoatman** [comment]
The short answer is yes; however most users wouldn't realize that they've switched apps unless they view the list of open apps.  We simply start an activity exposed by the broker rather than an activity that ships with MSAL... when interaction is required.  For acquireTokenSilent we invoke a bound service or call into the broker via AccountManager.

**Comment by NLLAPPS** [comment]
That would be great!

**Comment by mvdmee** [comment]
A might be a bit late with my feedback. But we just recently incorporated this library in our app to easy 2FA for our users.
And we ran into an problem with this "broker on by default" feature.
We found out that on most devices our app freezes when the Microsoft Authenticator app was installed on the same device (strangely enough, not all devices)
After some debugging we found out that it was looking for a broker app on the device and when it found one it resulted in a freeze. We still haven't found out why it freezes.
We ended up altering the source code and disabling the broker detection.
So, we where not so happy with the "on by default" feature :-)

BTW, is there a correct way of disabling the broker usage without changing the code?

**Comment by ashikns** [comment]
Seconding @mvdmee. To give some more info this code always returns an empty list (taken from msal source):
```
Intent microsoftAuthServiceIntent = new Intent("com.microsoft.identity.client.MicrosoftAuth");
microsoftAuthServiceIntent.setPackage(packageName);
microsoftAuthServiceIntent.setClassName(packageName, "com.microsoft.identity.client.MicrosoftAuthService");
List<ResolveInfo> infos = packageManager.queryIntentServices(microsoftAuthServiceIntent, 0);
return infos != null && infos.size() > 0;

**Comment by iambmelt** [comment]
@shoatman I am going to close this feedback request, as we've now shipped this functionality in `1.0.0`. Feel free to reopen if I have closed this in error. Thanks
